### PR TITLE
Add empty-template and empty-test-template rules

### LINKED ISSUE #443: [Rule] Empty Test Template
Empty test template can be used to overwrite suite test template. But if a) there is no test template in suite settigns b) we're using [Template] without more explicit NONE value we should report it.

### PR DESCRIPTION
**Pull request by bhirsz**
Closes #443 

For ``empty-template`` I have decided to not detect if there is ``Test Template`` set in ``*** Settings ***``. Any empty ``[Template]`` will report the same message (that there is empty setting and if you want to overwrite Test Template then use NONE).

### CODE REVIEW

File: docs/releasenotes/3.1.0.rst [hunk 1]
```diff
@@ -75,6 +75,12 @@ New W0918 ``multiline-inline-if`` rule that detects when inline IF is split into
 multiple lines. It's allowed, but it should be avoided, since it decreases readability.
 It recommends to use normal IF/ELSE instead.
 
+New rules that reports empty Template and Test Template (#443)
```
Comment (mnojek) [inline docs/releasenotes/3.1.0.rst:78]:
```suggestion
New rules that report empty Template and Test Template (#443)
```

File: docs/releasenotes/3.1.0.rst [hunk 2]
```diff
@@ -75,6 +75,12 @@ New W0918 ``multiline-inline-if`` rule that detects when inline IF is split into
...
+--------------------------------------------------------------
+
+Added new W0530 ``empty-template`` rule that reports ``[Template]`` setting without value.
+There is also new E0539 ``empty-test-template`` that reports empty ``Test Template``.
```
Comment (mnojek) [inline docs/releasenotes/3.1.0.rst:82]:
```suggestion
There is also new E0529 ``empty-test-template`` that reports empty ``Test Template``.
```

Comment (mnojek) [reply docs/releasenotes/3.1.0.rst:82]: Maybe let's also explain here why one has a warning level, and another one has error level of severity?

**Review by mnojek — APPROVED** [review]

File: docs/releasenotes/3.1.0.rst (hunk 2 above)
Comment (bhirsz) [reply docs/releasenotes/3.1.0.rst:82]: Yeah, I will do it

**Comment by bhirsz** [comment]
@mnojek I have added docs to ``empty-test-template`` to explain why it is error severity

File: robocop/checkers/lengths.py [hunk 3]
```diff
@@ -227,11 +227,22 @@
...
     ),
-    "0529": Rule(rule_id="0529", name="empty-test-template", msg="Test Template is empty", severity=RuleSeverity.ERROR),
+    "0529": Rule(
+        rule_id="0529",
+        name="empty-test-template",
+        msg="Test Template is empty",
+        docs="""
+        The ``Test Template`` set template to all tests in the suite. Empty value is considered an error
+        because it leads users to wrong impression on how the suite operates.
+        Without value, the setting is ignored and tests are not templated.
```
Comment (mnojek) [inline robocop/checkers/lengths.py:237]:
```suggestion
        ``Test Template`` sets the template to all tests in a suite. Empty value is considered an error
        because it leads the users to wrong impression on how the suite operates.
        Without value, the setting is ignored and the tests are not templated.
```

**Review by mnojek — APPROVED** [review]
